packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -0,0 +1,310 @@
+    setInputAmount(amount.replace(inputCurrencySymbol, ''))
+  }
+
+  function goToProvidersScreen() {
+    navigate(Screens.ProviderOptionsScreen, {
+      isCashIn: true,
+      currency: route.params.currency,
+      amount: localCurrencyAmount?.toNumber() || 0,
+    })
+  }
+
+  function onPressContinue() {
+    if (dollarAmount.isLessThan(DOLLAR_ADD_FUNDS_MIN_AMOUNT)) {
+      setShowingMinAmountDialog(true)
+      return
+    }
+    if (dollarAmount.isGreaterThan(dailyLimitCusd)) {
+      setShowingDailyLimitDialog(true)
+      return
+    }
+
+    goToProvidersScreen()
+  }
+
+  const closeMinAmountDialogAndContinue = () => {
+    setShowingDailyLimitDialog(false)
+    goToProvidersScreen()
+  }
+
+  const closeMinAmountDialogAndContact = () => {
+    setShowingDailyLimitDialog(false)
+    navigate(Screens.SupportContact, { prefilledText: t('dailyLimitRequest') })
+  }
+
+  return (
+    <SafeAreaView style={styles.container} edges={['bottom']}>
+      <Dialog
+        isVisible={showingMinAmountDialog}
+        actionText={t('minAmountDialog.dismiss')}
+        actionPress={closeMinAmountDialog}
+        testID={'MinAmountDialog'}
+      >
+        {t('minAmountDialog.body', { limit: `${currencySymbol}${minAmountInLocalCurrency}` })}
+      </Dialog>
+      <Dialog
+        isVisible={showingDailyLimitDialog}
+        actionText={t('dailyLimitDialog.continue')}
+        actionPress={closeMinAmountDialogAndContinue}
+        secondaryActionDisabled={false}
+        secondaryActionText={t('dailyLimitDialog.contact')}
+        secondaryActionPress={closeMinAmountDialogAndContact}
+        testID={'DailyLimitDialog'}
+      >
+        {
+          <Trans
+            i18nKey={'dailyLimitDialog.body'}
+            ns={Namespaces.fiatExchangeFlow}
+            tOptions={{ limit: `${dailyLimitCusd} cUSD`, contactEmail: CELO_SUPPORT_EMAIL_ADDRESS }}
+          >
+            <Text style={styles.emailLink} />
+          </Trans>
+        }
+      </Dialog>
+      <DisconnectBanner />
+      <KeyboardAwareScrollView
+        keyboardShouldPersistTaps={'always'}
+        contentContainerStyle={styles.contentContainer}
+      >
+        <View style={styles.amountInputContainer}>
+          <View>
+            <Text style={styles.exchangeBodyText}>
+              {isCusdCashIn ? t('global:amount') : t('amountCelo')}
+            </Text>
+          </View>
+          <TextInput
+            autoFocus={true}
+            keyboardType={'decimal-pad'}
+            onChangeText={onChangeExchangeAmount}
+            value={inputAmount.length > 0 ? `${inputCurrencySymbol}${inputAmount}` : undefined}
+            placeholderTextColor={colors.gray3}
+            placeholder={`${inputCurrencySymbol}0`}
+            style={[
+              styles.currencyInput,
+              isCusdCashIn ? styles.dollarCurrencyColor : styles.celoCurrencyColor,
+            ]}
+            testID="FiatExchangeInput"
+          />
+        </View>
+        {isCusdCashIn && (
```

Comment:
Shouldn't we also have an equivalent version of this for CELO cash-in? I think it would be equally, if not more, valuable to show people what the CELO conversion rate is before they buy

Reply:
I agree, there didn't use to be and there are no mockups for Celo, but I added the second line with what I thought was reasonable, added a screenshot to the PR description